### ISSUE TITLE
Language Specific Contexts

### ISSUE THREAD

**Issue by Remillard** [meetio-theme/sublime-meetio-theme#30]
I'd really appreciate it if you would make a conscious effort to stop scoping all the way to the language in a lot of cases.  I'm developing for VHDL and am in the process of rewriting the syntax.  During testing, I put in just a simple one for boolean "true".  White text.  I check the scope on the text and it is correct, `constant.language.vhdl`.  I think "Hey this is strange, I'm sure booleans in Python are covered".  I check Python code and lo and behold, "False" is colored.  I check scope and it's `constant.language.python`.

Seriously?  The scope rules are meant so that there is some consistency in these things.  I know every language is a little different, but what's the point of the scope naming document if theme makers ignore it.

TL;DR Please revisit your themes, and try to claw back all the standard scopes into your colors and remove the language specific portions of the scope names as much as possible.  Sure support special language features if you need to but `constant.language` (just an example, there are many others: see https://www.sublimetext.com/docs/3/scope_naming.html) is a basic scope that should be colored no matter what language is there.

**Comment by mauroreisvieira** [comment]
Hi @Remillard it's a good point, to resume that it's basically remove the specificity of scope right?

**Comment by mauroreisvieira** [comment]
I build this tool to help me to generate schemes for sublime, and I'm using it on Meetio.

https://github.com/meetio-theme/scheme-generator

My question is, for syntax like Vue or JSX/TSX, SCSS I still need the specificity right?

**Comment by Remillard** [comment]
Yes, basically your selectors go from most general to most specific with each accessor `.<term>`.  If you want a selector to work for only C+ comments you'd probably have something like `comment.block.cpp`.  If you wanted the same selector to work for ALL languages that had a block comment you'd just use `comment.block`, then any syntax writer that defines a comment block (e.g. `comment.block.vhdl`) will get your block color.

You're doing it already in a lot of places otherwise no other language would be colored, just in some really weird spots you have language specific colors when you should probably claw it back and say "hey this is common to all languages".

As for Vue/JSX/TSX/SCSS I don't know.  Honestly in my view you should **never** specify down to a language unless you know for certain there's a scope they use that no one else does.  Even then, you could still cut off the final language selector just in case someone someday uses that same syntax item.  By not specifying the language, you cover so many more languages than you would by trying to be specific.

**Comment by Remillard** [comment]
Looked at your generator.  The whole language specific thing is really backwards to the goal of using syntax language selectors.  That document I linked above has the scopes that are going to be used by 99% of languages with only mild variations between them.  Cover those FIRST then if there's some sort of oddball language specific one, then use the selector all the way down to the language and color that. 
 
Start general, adapt towards the specific instead of starting specific and leaving out everyone you didn't think of.  Saves you a lot of work 😄

**Comment by mauroreisvieira** [comment]
Thanks a lot for your feedback. It's optimize a lot my work.

**Comment by Remillard** [comment]
I look forward to seeing the update!

**Comment by mauroreisvieira** [comment]
It's more or like this: https://github.com/meetio-theme/scheme-generator/blob/master/src/rules/comment.ts

**Comment by Remillard** [comment]
I don't know exactly what I'm looking at, but perhaps it's your precursor to a color scheme.  Well those are language inspecific scope selectors so that's a good start.

My original comment was concerning `constant.language` but I figured there were a lot more in there.  And I have that other issue where I was adding things I thought might be interesting to identify `support.name` (which are basically language or library specific identifiers that have special meaning in the language). 

These are the scopes that language definers are going to be using: https://www.sublimetext.com/docs/3/scope_naming.html

**Comment by mauroreisvieira** [comment]
Hi @Remillard I'm finish the scopes: https://github.com/meetio-theme/sublime-meetio-theme/blob/master/schemes/Meetio-Theme-Dark.sublime-color-scheme

**Comment by Remillard** [comment]
I looked through it and I think it's an improvement.  It looks like you have very few language specific selectors, and I took some of those and went back and it looks like you have generic versions of those selectors too, so that's exactly the right way to do this (in my opinion.).  I didn't do a 100% check, but everything I checked looked like I would expect.  I hope the exercise helps.  If you end up producing a beta version of Mariana, I'd be happy to try it, see if I see anything oddball going back and forth between old and new.

**Comment by mauroreisvieira** [comment]
Hi @Remillard yes, during this week I will try to lunch a new version of Meetio Theme. 

Thanks for you feedback.
